tests/transport/rtps/publisher.cpp

```diff
@@ -345,7 +345,7 @@ int DDS_TEST::test(ACE_TString host, u_short port)
     control_sample.key = 0x04030201;
     DataSampleHeader dsh;
     dsh.message_id_ = INSTANCE_REGISTRATION;
-    dsh.sequence_ = SequenceNumber::SEQUENCENUMBER_UNKNOWN();
+    dsh.sequence_ = 2;
```

Comment:
Remind me again why we're modifying tests/transport/rtps files here? I thought you had your own test for these changes in tests/transport/best_effort_reader. Did your changes break this test?

Reply:
The tests/transport/rtps used SequenceNumber::SEQUENCENUMBER_UNKNOWN() to send 4 control messages. That fails the sequence checking implemented in this PR. I had thought about creating a separate PR for tests/transport/rtps, but took a shortcut and made the changes in this PR, thinking that’s the same issue this PR is for. I can create a  separate PR for tests/transport/rtps, if that is what we prefer.